app/controllers/application_controller.rb

```diff
@@ -1,5 +1,14 @@
 class ApplicationController < ActionController::Base
-  def current_user
-    User.first
+  protect_from_forgery with: :exception
+
+  before_action :update_allowed_parameters, if: :devise_controller?
+
```

Comment:
- [ ] Despite installing devise I am still able to visit several supposedly unauthorized routes, like the new post route. Kindly ensure that only logged-in users can use the app by adding the code below to this file:

  ```
  before_action :authenticate_user!
  ```